recipes/crunch/all/CMakeLists.txt

```diff
@@ -0,0 +1,235 @@
+    ${CRUNCH_FOLDER}/crnlib/crn_threaded_clusterizer.h
+    ${CRUNCH_FOLDER}/crnlib/crn_threaded_resampler.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_threaded_resampler.h
+    ${CRUNCH_FOLDER}/crnlib/crn_threading.h
+    ${CRUNCH_FOLDER}/crnlib/crn_threading_null.h
+    ${CRUNCH_FOLDER}/crnlib/crn_timer.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_timer.h
+    ${CRUNCH_FOLDER}/crnlib/crn_traits.h
+    ${CRUNCH_FOLDER}/crnlib/crn_tree_clusterizer.h
+    ${CRUNCH_FOLDER}/crnlib/crn_types.h
+    ${CRUNCH_FOLDER}/crnlib/crn_utils.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_utils.h
+    ${CRUNCH_FOLDER}/crnlib/crn_value.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_value.h
+    ${CRUNCH_FOLDER}/crnlib/crn_vec.h
+    ${CRUNCH_FOLDER}/crnlib/crn_vec_interval.h
+    ${CRUNCH_FOLDER}/crnlib/crn_vector.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_vector.h
+    ${CRUNCH_FOLDER}/crnlib/crn_vector2d.h
+    ${CRUNCH_FOLDER}/crnlib/crn_winhdr.h
+    ${CRUNCH_FOLDER}/crnlib/crn_zeng.cpp
+    ${CRUNCH_FOLDER}/crnlib/crn_zeng.h
+    ${CRUNCH_FOLDER}/crnlib/crnlib.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zBuf.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zBuf.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zBuf2.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zCrc.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zFile.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zFile.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zStream.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_7zVersion.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_Alloc.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_Alloc.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_Bcj2.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_Bcj2.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_Bra.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_Bra.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_Bra86.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_BraIA64.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_CpuArch.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzFind.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzFind.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzHash.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaDec.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaDec.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaEnc.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaEnc.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaLib.cpp
+    ${CRUNCH_FOLDER}/crnlib/lzma_LzmaLib.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_MyVersion.h
+    ${CRUNCH_FOLDER}/crnlib/lzma_Types.h
+    ${CRUNCH_FOLDER}/crnlib/../inc/crn_decomp.h
+    ${CRUNCH_FOLDER}/crnlib/../inc/crnlib.h
+    ${CRUNCH_FOLDER}/crnlib/../inc/dds_defs.h
+)
+
+if(WIN32)
+    list(APPEND CRNLIB_SRCS
+        ${CRUNCH_FOLDER}/crnlib/lzma_LzFindMt.cpp
+        ${CRUNCH_FOLDER}/crnlib/lzma_LzFindMt.h
+        ${CRUNCH_FOLDER}/crnlib/lzma_Threads.cpp
+        ${CRUNCH_FOLDER}/crnlib/lzma_Threads.h
+        ${CRUNCH_FOLDER}/crnlib/crn_threading_win32.cpp
+        ${CRUNCH_FOLDER}/crnlib/crn_threading_win32.h
+    )
+else(WIN32)
+    list(APPEND CRNLIB_SRCS
+        ${CRUNCH_FOLDER}/crnlib/crn_threading_pthreads.h
+        ${CRUNCH_FOLDER}/crnlib/crn_threading_pthreads.cpp
+    )
+endif()
+
+add_library(crnlib ${CRNLIB_SRCS})
```

Comment:
GDAL expects `crunch` as libname. Are you sure that crnlib is official libname in upstream build files?

Reply:
Yes https://github.com/BinomialLLC/crunch/blob/master/crnlib/crnlib.2008.vcproj

Reply:
GDAL uses a fork of crunch that added CMake build files. https://github.com/OSGeo/gdal/blob/master/gdal/doc/source/drivers/raster/dds.rst#build-instructions. This fork defines the lib as `crunch` https://github.com/rouault/crunch/blob/build_fixes/crnlib/CMakeLists.txt#L91

Reply:
ok, another patch for `configure.ac`'s gdal recipe :sweat_smile: